openbsd: init at 7.5

ali-abrar opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Add OpenBSD support similar to FreeBSD/NetBSD. We use some NetBSD tools to help bootstrap. Currently these are known o build:

```
nix-build -A pkgsCross.x86_64-openbsd.hello
nix-build -A pkgsCross.x86_64-openbsd.zlib
```

Actually *running* these OpenBSD executables on OpenBSD will take more work, but @ericson2314 thinks that is fine for now, because it is easier to fix that after there is a building foundation.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ericson2314 committed:
netbsd.makeMinimal: Make it possible to override the rules

Co-authored-by: Rahul Butani <rrbutani@users.noreply.github.com>

Ericson2314 committed:
llvmPackages.libcxx: Fix compiling for OpenBSD

Ericson2314 committed:
llvmPackages.clang: Fix special-case for OpenBSD to be less special

ali-abrar committed:
llvmPackages: libcxx: link stdlib

See: https://github.com/NixOS/nixpkgs/pull/246577/files#r1600650238

This code was breaking some thing for me, and I believe it was a
mistake. I hink we *do* wanna link libc, since libc++ depends on it, but
*don't* want to link an already-existing C++ standard library
implementation. `-nostdlib++` without `-nostdlib` therefore seems
correct to me.

Without this change, we do indeed fail to link OpenBSD.

Ericson2314 commented on pkgs/development/compilers/llvm/17/default.nix:
```diff
@@ -150,6 +150,9 @@ in let
         })
+      ] ++ lib.optionals stdenv.targetPlatform.isOpenBSD [
+        # TODO make unconditional
+        ../common/clang/openbsd-no-assume-path-exists.patch
```
https://github.com/llvm/llvm-project/pull/92183

Ericson2314 replied:
@RossComputerGuy I suppose maybe when they are merged to master, we can backport to at least 18 and switch to using that for this?

RossComputerGuy replied:
@Ericson2314 Sounds good to me.

Ericson2314 replied:
https://github.com/llvm/llvm-project/pull/92293 is a backport PR (that unlike the original I cannot merge). We can cherry-pick instead for extra precision for LLVM 18.

ali-abrar replied:
Switched to LLVM 18 with these patches.

Ericson2314 commented on pkgs/development/compilers/llvm/17/default.nix:
```diff
@@ -405,6 +408,11 @@ in let
+        # See https://lists.boost.org/boost-bugs/2015/07/41690.php and
+        # https://github.com/boostorg/log/commit/c98b1f459add14d5ce3e9e63e2469064601d7f71
+        # for a description of an analogou problem an fix for Boost.
+        ../common/libcxx/openbsd.patch
```
https://github.com/llvm/llvm-project/pull/92186

ali-abrar replied:
This is no longer present in 17, moved to 18 and fetched from the linked PR instead of using an in-tree patch.

Ericson2314 commented on pkgs/os-specific/bsd/openbsd/pkgs/make-rules/ar.patch:
Submitted an email to `tech@openbsd.org`. Hopefully it will appear shortly.

Ericson2314 replied:
https://marc.info/?l=openbsd-tech&m=171575284906018&w=2

alyssais replied:
Great! Let's fetchurl this from https://marc.info/?l=openbsd-tech&m=171575284906018&q=raw.

rhelmot commented on pkgs/os-specific/bsd/openbsd/pkgs/libc/package.nix:
```diff
@@ -0,0 +1,117 @@
+
+    make -C $BSDSRCDIR/lib/libutil $makeFlags
+    make -C $BSDSRCDIR/lib/libutil $makeFlags install
+    symlink_so util
```
What packages have you tried compiling with this libc? I found that there were actually a bunch more libraries on freebsd which were required pretty much constantly, once you got into more interesting applications, like rust packages and stuff built with meson.

Ericson2314 replied:
Just the basics, we will also need to add more stuff

rhelmot commented on pkgs/os-specific/bsd/netbsd/pkgs/make-rules.nix:
```diff
@@ -0,0 +1,75 @@
+
+mkDerivation {
+  path = "share/mk";
+  sha256 = "0w9x77cfnm6zwy40slradzi0ip9gz80x6lk7pvnlxzsr2m5ra5sy";
```
does this hash need to be here?

Ericson2314 replied:
It does because that's NetBSD, which has many separate fetches (probably not good).

rhelmot commented on pkgs/os-specific/bsd/openbsd/pkgs/mkDerivation.nix:
```diff
@@ -0,0 +1,91 @@
+      ];
+
+      # amd64 not x86_64 for this on unlike NetBSD
+      MACHINE_ARCH = "amd64";
```
You may, but are not required to, make this more generic sooner rather than later. I've heard openbsd has the best hardware support for other architectures out of all the bsds!

rhelmot reviewed (commented):
no showstoppers, just some questions and advice from the freebsd mines

Ericson2314 commented on pkgs/os-specific/bsd/openbsd/pkgs/include/package.nix:
```diff
@@ -0,0 +1,46 @@
+  ];
+
+  makeFlags = [
+    "RPCGEN_CPP=${buildPackages.stdenv.cc.cc}/bin/cpp"
```
This build-time CPP (not that it really matters) is not on the path

Ericson2314 commented on pkgs/os-specific/bsd/openbsd/pkgs/include/package.nix:
```diff
@@ -0,0 +1,46 @@
+
+  extraPaths = [
+    "lib"
+    "sys"
```
It is just some dirs now

alyssais commented on pkgs/development/compilers/llvm/common/libcxx/openbsd.patch:
```diff
@@ -0,0 +1,127 @@
+    
+      See https://lists.boost.org/boost-bugs/2015/07/41690.php and
+      https://github.com/boostorg/log/commit/c98b1f459add14d5ce3e9e63e2469064601d7f71
+      for a description of an analogou problem an fix for Boost.
```
```suggestion
      for a description of an analogous problem an fix for Boost.
```

Ericson2314 replied:
(We'll be replacing this with `fetchpatch` of https://github.com/llvm/llvm-project/pull/92186)

ali-abrar replied:
The typo also exists in pkgs/development/compilers/llvm/17/default.nix, and I've fixed it there.

ali-abrar replied:
We're now using fetchpatch here.

alyssais commented on pkgs/os-specific/bsd/netbsd/pkgs/make-rules.nix:
```diff
@@ -0,0 +1,75 @@
+  nativeBuildInputs = [
+    bsdSetupHook
+    netbsdSetupHook
+    rsync
```
What's rsync used for?

Ericson2314 replied:
NetBSD's combining the extra paths, which is not as good as what we did for FreeBSD and OpenBSD. We should really switch it over to the same way of doing things as the others at some point.

ali-abrar replied:
I'd like to do that in a separate PR.

Ericson2314 replied:
Oh yes, I meant that. It is definitely out of scope for this one,

alyssais commented on pkgs/os-specific/bsd/netbsd/pkgs/make.nix:
```diff
@@ -11,53 +12,12 @@ mkDerivation {
+    make-rules.postPatch
+    + ''
+      # make needs this to pick up our sys make files
+      export NIX_CFLAGS_COMPILE+=" -D_PATH_DEFSYSPATH=\"$out/share/mk\""
```
appendToVar?

alyssais commented on pkgs/development/compilers/llvm/17/default.nix:
```diff
@@ -150,6 +150,9 @@ in let
           libllvmLibdir = "${tools.libllvm.lib}/lib";
         })
+      ] ++ lib.optionals stdenv.targetPlatform.isOpenBSD [
+        # TODO make unconditional
```
Is this a promise to submit a PR to staging making it unconditional as soon as this is merged to staging?

Ericson2314 replied:
If we switch to LLVM 18 perhaps it doesn't even need to go to staging!

alyssais commented on pkgs/development/compilers/llvm/17/default.nix:
```diff
@@ -408,6 +408,11 @@ in let
           url = "https://github.com/macports/macports-ports/raw/acd8acb171f1658596ed1cf25da48d5b932e2d19/lang/llvm-17/files/0042-mbstate_t-not-defined.patch";
           hash = "sha256-jo+DYA6zuSv9OH3A0bYwY5TlkWprup4OKQ7rfK1WHBI=";
         })
+      ] ++ lib.optionals stdenv.hostPlatform.isOpenBSD [
```
Why conditional?

Ericson2314 replied:
Just to avoid rebuilds. Like before I hope we can avoid that problem by switching to 18.

alyssais commented on pkgs/development/compilers/llvm/common/libcxx/default.nix:
```diff
@@ -87,12 +87,6 @@ let
-    # but that does not appear to be the case for example when building
-    # pkgsLLVM.libcxxabi (which uses clangNoCompilerRtWithLibc).
-    "-DCMAKE_EXE_LINKER_FLAGS=-nostdlib"
-    "-DCMAKE_SHARED_LINKER_FLAGS=-nostdlib"
```
Can this please be explained in the commit message rather than just a GitHub link that may disappear?

Ericson2314 replied:
Done

alyssais commented on pkgs/development/compilers/llvm/common/compiler-rt/default.nix:
```diff
@@ -149,7 +150,9 @@ stdenv.mkDerivation ({
-     ln -s $out/lib/freebsd/libclang_rt.builtins-*.a $out/lib/libgcc.a
+     ln -s $out/lib/*/libclang_rt.builtins-*.a $out/lib/libgcc.a
+  '' + lib.optionalString forceLinkCompilerRt ''
+     ln -s $out/lib/*/libclang_rt.builtins-*.a $out/lib/libcompiler_rt.a
```
Any idea why we need this?

Ericson2314 replied:
I think the build system simply doesn't bother to make the "normally named" library when its only statically linking? I am a bit confused why too. But note how the previous hack also had to work from the more obscure `libclang_rt.builtins`.

Ericson2314 replied:
I suppose we can add a comment explaining this

Ericson2314 replied:
Added those comments

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/include/no-perms.patch:
```diff
@@ -0,0 +1,12 @@
+ 		echo installing in ${.CURDIR}/$$i; \
+ 		cd ${.CURDIR}/$$i && ${RUN_MAKE}; \
+ 	done
+-	chown -RP ${BINOWN}:${BINGRP} ${DESTDIR}/usr/include
```
Could we avoid the patch by setting BINOWN and BINGRP to the current user and group in a preBuild?

ali-abrar replied:
fixed

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/libc/disable-librebuild.patch:
```diff
@@ -0,0 +1,12 @@
+ .include <bsd.own.mk>
+
+ LIB=c
+-LIBREBUILD=y
```
We can't override this from the make command line?

Ericson2314 replied:
it unfortunately just checks that the variable is defined, not that the variable value is `y`.

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/libc/libc-no-tags.patch:
```diff
@@ -0,0 +1,24 @@
+ 
++all: ${SRCS}
++
++LIBC_TAGS ?= yes
```
Why `?=`?

Ericson2314 replied:
To hint that it's just a default. I vaguely recall `?=` isn't as useful for BSD Make as it is for GNU Make, but I forget the details.

alyssais replied:
Every make variable that doesn't use := is just a default.  The only different the ? makes is that it will prefer a value set in the environment (rather than as a make flag) to the default, which IMO is quite fragile unless it's a conventional environment variable like CC.

Ericson2314 replied:
Ah right, thanks!

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/libc/libpthread-no-werror.patch:
```diff
@@ -0,0 +1,13 @@
+ LIBCSRCDIR=    ${.CURDIR}/../libc
+
+-CFLAGS+=-Wall -g -Werror -Wshadow
++CFLAGS+=-Wall -g -Wshadow
```
Could we use NIX_CFLAGS_COMPILE rather than a patch?

Ericson2314 replied:
With `-Wno-error`?

alyssais replied:
Yeah.

ali-abrar replied:
done

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/libc/package.nix:
```diff
@@ -0,0 +1,117 @@
+  ];
+
+  patches = [
+    # NetBSD's make usess `${variable:tl}` not `${variable:L}`.
```
Could we move this into the patch file (everything at the start is interpreted as a comment) so the explanation lives closer to the code change?

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/libc/package.nix:
```diff
@@ -0,0 +1,117 @@
+    csu
+  ];
+
+  env.NIX_CFLAGS_COMPILE = builtins.toString [ "-B${csu}/lib" ];
```
Why make a single element list and then convert it back to a string?

Ericson2314 replied:
There used to be more things in the list :sweat_smile:

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/make-rules/package.nix:
```diff
@@ -0,0 +1,28 @@
+  noCC = true;
+
+  buildInputs = [ ];
+  nativeBuildInputs = [ ];
```
Are these lines necessary?

Ericson2314 replied:
Yes because there are default deps to be cancelled.

alyssais reviewed (changes requested):
First, thank you so much for splitting this into small, easily reviewable commits.

How come we're using NetBSD make and working around its differences instead of just building OpenBSD make?

Ericson2314 commented:
@alyssais 

> How come we're using NetBSD make and working around its differences instead of just building OpenBSD make?

OpenBSD doesn't have any `tools/compat` or similar way to build its programs with host platform != OpenBSD. So rather than fight hard to build the `nativeBuildInputs` we'd need, we just cut the Gordian knot and used NetBSD ones.

I am still chuckling about this :)

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/mkDerivation.nix:
```diff
@@ -0,0 +1,99 @@
+
+      meta = with lib; {
+        maintainers = with maintainers; [ ericson2314 ];
+        platforms = platforms.unix;
```
> OpenBSD doesn't have any tools/compat or similar way to build its programs with host platform != OpenBSD.

In that case, why are we aren't we defaulting this to OpenBSD?

Ericson2314 replied:
Good catch! :)

ali-abrar committed:
openbsd: init at 7.5

Ericson2314 committed:
llvmPackages.compiler-rt: Add flag to force libcompiler-rt.a creation

See the code comments for details

Also replace a bad hard-coded `freebsd` with a flexible `*` wildcard in
the `doFakeLibgcc` implementation.

Ericson2314 commented:
OK all review items are addressed!

rrbutani commented on pkgs/os-specific/bsd/netbsd/pkgs/make-rules.nix:
```diff
@@ -0,0 +1,75 @@
+       substituteInPlace $BSDSRCDIR/share/mk/bsd.kinc.mk \
+         --replace /bin/rm rm
+    ''
+    + lib.optionalString stdenv.targetPlatform.isDarwin ''
```
Was changing this from `stdenv.isDarwin` to `stdenv.targetPlatform.isDarwin` intentional?

Ericson2314 replied:
The make rules are always a build-time input to something, so I don't really get the sense of getting in the host platform of the rules themselves (i.e. the build platform of the thing to be built).

Ericson2314 replied:
I don't know the purpose of this bit though, which doesn't help either.

rrbutani commented on pkgs/os-specific/bsd/netbsd/pkgs/makeMinimal.nix:
```diff
@@ -22,7 +23,9 @@ mkDerivation {
-    ${make.postPatch}
+
+    # make needs this to pick up our sys make files
+    export NIX_CFLAGS_COMPILE+=" -D_PATH_DEFSYSPATH=\"$out/share/mk\""
```
Can we use `appendToVar` here as well, for consistency?

```suggestion
    appendToVar NIX_CFLAGS_COMPILE -D_PATH_DEFSYSPATH="$out/share/mk"
```

rrbutani commented on pkgs/os-specific/bsd/openbsd/pkgs/libc/package.nix:
```diff
@@ -0,0 +1,120 @@
+
+  # Suppress lld >= 16 undefined version errors
+  # https://github.com/freebsd/freebsd-src/commit/2ba84b4bcdd6012e8cfbf8a0d060a4438623a638
+  env.NIX_LDFLAGS = lib.optionalString (stdenv.targetPlatform.linker == "lld") "--undefined-version";
```
Apologies if I'm missing something; should this be `hostPlatform`?

Ericson2314 replied:
Yes it should

rrbutani reviewed (commented):
The LLVM changes all look good to me now. I'm not really qualified to review the rest but I had a couple of questions:

Ericson2314 commented on pkgs/os-specific/bsd/netbsd/pkgs/make.nix:
```diff
@@ -11,53 +12,12 @@ mkDerivation {
+    make-rules.postPatch
+    + ''
+      # make needs this to pick up our sys make files
+      appendToVar NIX_CFLAGS_COMPILE -D_PATH_DEFSYSPATH="$out/share/mk"
```
```suggestion
      appendToVar NIX_CFLAGS_COMPILE -D_PATH_DEFSYSPATH="$out/share/mk"
```
```suggestion
      appendToVar NIX_CFLAGS_COMPILE "-D_PATH_DEFSYSPATH=$out/share/mk"
```

alyssais commented on pkgs/development/compilers/llvm/18/default.nix:
```diff
@@ -146,6 +146,11 @@ in let
         ./clang/gnu-install-dirs.patch
         ../common/clang/add-nostdlibinc-flag.patch
+        (fetchpatch {
+          url = "https://github.com/llvm/llvm-project/commit/5b09e6bf5aaeb21256e86455f8d6c42d01f71b95.patch";
```
Shouldn't this be https://github.com/llvm/llvm-project/commit/48c1364200b5649dda2f9ccbe382b0bd908b99de, the commit that actually exists in that repo?

Would be nice for the patch to have a name too, so it's obvious what it's for.

ali-abrar replied:
this is done

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/source.nix:
```diff
@@ -0,0 +1,8 @@
+  repo = "src";
+  rev = "19e99d061340840dc52e10ae9c02d78a0a455689";
+  sha256 = "1061vb5m7k7q502bqsfrnnszjv9v66m0ksg5jka5jsdn6qa0fd9n";
+}
```
I think we should fetch from the canonical CVS repo, not the GitHub mirror.

Ericson2314 replied:
We're having trouble fetching this, e.g. 

```bash
cvs -d :pserver:anoncvs@anoncvs4.usa.openbsd.org:/cvs checkout -rOPENBSD_7_5 -P src
```
(with a shell with `openssh` `cvs` `inetutils`) is not working.

alyssais replied:
This looks like it's working for me:

```nix
fetchcvs {
  cvsRoot = "anoncvs@anoncvs.fr.openbsd.org/cvs";
  module = "src";
  tag = "OPENBSD_7_5";
  sha256 = lib.fakeSha256;
}
```

In a shell you'd have to set CVS_RSH=ssh to match the behaviour of fetchcvs.

But actually, why can't we just use the source tarball they publish?

https://cdn.openbsd.org/pub/OpenBSD/7.5/src.tar.gz

Ericson2314 replied:
> In a shell you'd have to set CVS_RSH=ssh to match the behaviour of fetchcvs.

oh!

> But actually, why can't we just use the source tarball they publish?

Oh great. I like the look of "CDN". Yes let's do that.

alyssais replied:
Probably wouldn't be too bad using `srcs`, but this is also fine.

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/mkDerivation.nix:
```diff
@@ -0,0 +1,91 @@
+    }
+    // lib.optionalAttrs stdenv'.hasCC {
+      # TODO should CC wrapper set this?
+      CPP = "${stdenv'.cc.targetPrefix}cpp";
```
No real need for this to be conditional, right?

ali-abrar replied:
fixed

Ericson2314 replied:
Actually yes, because if we don't have a CC then we don't have a `stdenv'.cc.targetPrefix`. And trying to be clever about this can introduce infinite recursion.

alyssais commented on pkgs/development/compilers/llvm/18/default.nix:
```diff
@@ -396,7 +396,12 @@ in let
-      patches = lib.optionals (stdenv.isDarwin && lib.versionOlder stdenv.hostPlatform.darwinMinVersion "10.13") [
+      patches = [
+        (fetchpatch {
+          url = "https://github.com/llvm/llvm-project/commit/24ef89b6a3c2513221dc2e154642ce379cd701f6.patch";
```
Similarly I think this should be https://github.com/llvm/llvm-project/commit/1118c2e05e67a36ed8ca250524525cdb66a55256.

ali-abrar replied:
done

Ericson2314 reviewed (approved):
I have a feeling moving from Git is not worth it.

alyssais commented on pkgs/os-specific/bsd/openbsd/pkgs/source.nix:
```diff
@@ -0,0 +1,8 @@
+{ lib, fetchcvs }:
```
Presumably forgot to remove this after using `lib.fakeSha256`? :)

afh commented:
Glad to see this PR has been merged, thank you for your work, @ali-abrar, can't wait to try it out! 💯

foolnotion commented:
Sorry but this seems to break `clang-18.1.6` on linux:

```
❯ nix log /nix/store/synw5z7kjmp7gmg4knh4ikaffprbzr3m-clang-18.1.6.drv^\*
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/pkdznhza3cbgdixj2czjw6qi3v8wfpir-clang-src-18.1.6
source root is clang-src-18.1.6/clang
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
applying patch /nix/store/4qh1l55wx654v0ls51dg1wdi5xy93w9b-purity.patch
patching file lib/Driver/ToolChains/Gnu.cpp
applying patch /nix/store/fk8qk07cl9m7wkjb0xc0hphbh3vrxfqh-gnu-install-dirs.patch
patching file cmake/modules/AddClang.cmake
patching file lib/Headers/CMakeLists.txt
Hunk #1 succeeded at 498 (offset 25 lines).
patching file tools/libclang/CMakeLists.txt
Hunk #1 succeeded at 246 (offset 12 lines).
patching file tools/scan-build-py/CMakeLists.txt
applying patch /nix/store/gxqzh9x4wmiv4jpbvla3z7i26zkkacqd-add-nostdlibinc-flag.patch
patching file lib/Driver/Driver.cpp
Hunk #1 succeeded at 499 (offset 8 lines).
applying patch /nix/store/ycw68pzfjx00fm99rqlvfnn7hssnzakc-48c1364200b5649dda2f9ccbe382b0bd908b99de.patch
patching file lib/Driver/ToolChains/OpenBSD.cpp
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file lib/Driver/ToolChains/OpenBSD.cpp.rej
```

alyssais commented:
Ah I wonder if that's because we updated LLVM and now the patch is already there and doesn't need backporting any more.

paparodeo commented:
patches reverted in: https://github.com/NixOS/nixpkgs/pull/315790
patches were the last 2 changes applied before 18.1.6 was tagged.
https://github.com/llvm/llvm-project/commits/llvmorg-18.1.6/

Ericson2314 commented:
Sorry about that!
